homeassistant/components/ecobee/__init__.py

```diff
@@ -116,6 +125,26 @@ def __init__(self, hass, entry, api_key, refresh_token):
         _LOGGER.error("Error updating ecobee tokens")
         return False
 
+    async def request(self, method, *args, **kwargs):
+        """Wrap calls to the ecobee API to catch token errors and retry."""
+        req = functools.partial(method, *args, **kwargs)
+        try:
+            return await self._hass.async_add_executor_job(req)
+        except ExpiredTokenError:
+            _LOGGER.warning(
+                "Ecobee request failed due to expired token; refreshing and retrying"
+            )
+            if await self.refresh():
+                return await self._hass.async_add_executor_job(req)
+            _LOGGER.error(
+                "Ecobee request failed due to expired token and token could not be refreshed"
+            )
+        except InvalidTokenError:
+            _LOGGER.error(ECOBEE_INVALID_TOKEN_MESSAGE)
+            self._hass.components.persistent_notification.async_create(
```

Comment:
Let's just log an error. We don't need a persistent notification too.

Reply:
My only thought with persistent notification was that the ecobee integration will stop working, and so the persistent notification would be a more visible way of alerting the user why. But I am happy to remove if you prefer.

Reply:
We normally don't use the persistent notification anymore. I don't think the user experience is so much better with the notification.

It would be better if we can easily allow the user to get new tokens, eg by starting the config flow again via import. The user would be notified and would have a way of resolving the issue.The problem there is that we currently don't have a way of retaining config entry customizations from an existing config entry if we start a new flow for the corresponding config.